### PR TITLE
Patchnamesuggestion d7

### PR DESCRIPTION
**Pull request by helmo**
Updates for #47 en #55

### CODE REVIEW

**Comment by markhalliwell** [comment]
@clemens-tolboom, I really do appreciate your work on this :) I am going to wait until we actually get JSON support on d.o before merging this in though. We will likely have to do more releases before JSON support is actually added.

**Comment by clemens-tolboom** [comment]
I'm not sure we want to use JSON at all. RestWS only deliver entities which renders most dreditor usages request intense. For creating a commit message we need the issue node + all comments + all participants

It probably is better to have a cache mechanism of the HTML issue at hand to use for generating commit message, comments count.

Both dreditor and drush_iq needs these metrics.

See also
- d.o [Deploy RestWS for D7 project issue JSON](https://drupal.org/node/1710850)
- drush_iq [Fix for the new json structure on d.o D7](https://drupal.org/node/2127965)

**Comment by clemens-tolboom** [comment]
Btw this patch is from @helmo and I'll discuss my previous remark with @helmo this week.

**Comment by helmo** [comment]
Here's a version bases on the new reality of split files.

**Comment by helmo** [comment]
This is a temporary solution until the issue (https://drupal.org/node/2125269 - [d7qa] [Bug report] [META] Regressions in issue queue workflow) is fixed.  I guess that most elements will be back on the same page then.

File: src/js/plugins/issue.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ Drupal.dreditor.issue.getNid = function() {
  */
 Drupal.dreditor.issue.getNewCommentNumber = function() {
   // Get comment count.
-  return parseInt($('#comment-form .comment-inner > h3').text().match(/\d+$/)[0], 10);
+  // TODO find a D7 replacement
```
Comment (markhalliwell) [inline src/js/plugins/issue.js]: We could always do an AJAX call to get the contents of the node and then traverse the response to get this (temporarily, until JSON support is added).

**Comment by clemens-tolboom** [comment]
I have code for this in my sandbox. This will probably be available next Friday. @helmo that's about the stock response fetching and cache.

**Comment by markhalliwell** [comment]
I'm going to close this PR. We'll likely want to abstract a lot of the JSON layers in Dreditor once d.o supports it. Also, given the new structure and #106, I suspect we'll move into a more model/dispatcher mode to implement this.

**Pull request closed without merging**